File: app/src/main/java/com/example/pachinui2/ScoreAdapter.java
package com.example.pachinui2;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.lang.reflect.Array;
import java.util.ArrayList;

public class ScoreAdapter extends RecyclerView.Adapter<ScoreAdapter.ScoreHandler> {


    ArrayList<ScoreModel> list;

    public ScoreAdapter(ArrayList<ScoreModel> list) {
        this.list = list;
    }


    @Override
    public ScoreHandler onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_adv,parent,false);
        ScoreHandler scoreHandler = new ScoreHandler(view);
        return scoreHandler;
    }

    @Override
    public void onBindViewHolder(ScoreAdapter.ScoreHandler holder, int position) {

        ScoreModel scoreModel = list.get(position);
        holder.imageViewScore.setImageResource(scoreModel.getImageUri());


    }

    @Override
    public int getItemCount() {
        return list.size();
    }

    class ScoreHandler extends RecyclerView.ViewHolder{
        ImageView imageViewScore;

        public ScoreHandler(@NonNull View itemView){
            super(itemView);

            imageViewScore = itemView.findViewById(R.id.iv_movie);

        }
    }
}

File: app/src/main/java/com/example/pachinui2/ScoreAdapter.kt
package com.example.pachinui2

import com.example.pachinui2.ScoreModel
import androidx.recyclerview.widget.RecyclerView
import com.example.pachinui2.ScoreAdapter.ScoreHandler
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import android.widget.ImageView
import com.example.pachinui2.R
import java.util.ArrayList

class ScoreAdapter(var list: ArrayList<ScoreModel>) : RecyclerView.Adapter<ScoreHandler>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ScoreHandler {
        val view =
            LayoutInflater.from(parent.context).inflate(R.layout.item_adv, parent, false)
        return ScoreHandler(view)
    }

    override fun onBindViewHolder(holder: ScoreHandler, position: Int) {
        val scoreModel = list[position]
        holder.imageViewScore.setImageResource(scoreModel.imageUri)
    }

    override fun getItemCount(): Int {
        return list.size
    }

    inner class ScoreHandler(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var imageViewScore: ImageView

        init {
            imageViewScore = itemView.findViewById(R.id.iv_movie)
        }
    }
}